src/ui/mainwindow.cpp

```diff
@@ -104,9 +104,13 @@ MainWindow::MainWindow(Core::Application *app, QWidget *parent) :
             ui->lineEdit, static_cast<void (SearchEdit::*)()>(&SearchEdit::setFocus));
 
     restoreGeometry(m_settings->windowGeometry);
-    ui->splitter->restoreState(m_settings->splitterGeometry);
+    ui->splitter->restoreState(m_settings->verticalSplitterGeometry);
     connect(ui->splitter, &QSplitter::splitterMoved, [=](int, int) {
-        m_settings->splitterGeometry = ui->splitter->saveState();
+        m_settings->verticalSplitterGeometry = ui->splitter->saveState();
+    });
+    connect(ui->sectionsSplitter, &QSplitter::splitterMoved, [=](int, int) {
+        if (ui->sections->isVisible())
```

Comment:
How about storing the height of `sections` in `displaySections()` before hiding it?